src/progress.rs

```diff
@@ -50,6 +53,92 @@ impl<'a> Progress<'a> {
         )
         .await
     }
+
+    /// Wait until MeiliSearch processes an update, and get its status.
+    /// 
+    /// interval_ms = The frequency at which the server should be polled in milliseconds. Default = 50ms
+    /// timeout_ms = The maximum time to wait for processing to complete in milliseconds. Default = 5000ms
+    /// 
+    /// # Example
+    ///
+    /// ```
+    /// # use meilisearch_sdk::{client::*, document, indexes::*, progress::*};
+    /// # use serde::{Serialize, Deserialize};
+    /// 
+    /// # #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    /// # struct Document {
+    /// #    id: usize,
+    /// #    value: String,
+    /// #    kind: String,
+    /// # }
+    /// # 
+    /// # impl document::Document for Document {
+    /// #    type UIDType = usize;
+    /// #
+    /// #    fn get_uid(&self) -> &Self::UIDType {
+    /// #        &self.id
+    /// #    }
+    /// # }
+    /// 
+    /// # futures::executor::block_on(async move {
+    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// let movies = client.create_index("movies_wait_for_pending", None).await.unwrap();
+    /// 
+    /// let progress = movies.add_documents(&[
+    ///     Document { id: 0, kind: "title".into(), value: "The Social Network".to_string() },
+    ///     Document { id: 1, kind: "title".into(), value: "Harry Potter and the Sorcerer's Stone".to_string() },
+    /// ], None).await.unwrap();
+    /// 
+    /// let status = progress.wait_for_pending_update(None, None).await.unwrap();
+    /// 
+    /// # client.delete_index("movies_wait_for_pending").await.unwrap();
+    /// assert!(matches!(status, UpdateStatus::Processed { .. }));
+    /// # });
+    /// ```
+    pub async fn wait_for_pending_update(
+        &self,
+        interval_ms: Option<u64>,
+        timeout_ms: Option<u64>,
+    ) -> Result<UpdateStatus, Error> {
+        let interval: u64;
+        let timeout: u64;
+
+        match interval_ms {
+            Some(v) => interval = v,
+            None => interval = 50,
+        }
+
+        match timeout_ms {
+            Some(v) => timeout = v,
+            None => timeout = 5000,
+        }
+
+        let mut elapsed_time = 0;
+        let mut status: UpdateStatus;
+
+        while timeout > elapsed_time {
+            status = self.get_status().await?;
+
+            match status {
+                UpdateStatus::Failed { .. } | UpdateStatus::Processed { .. } => {
+                    return self.get_status().await;
+                },
+                UpdateStatus::Enqueued { .. } => {
+                    elapsed_time += interval;
+                    sleep(Duration::from_millis(interval));
+                },
+            };
+        }
+
+        Err(
```

Comment:
It would be better to return None in the case of a timeout. It's much easier to handle. Additionally, a timeout is a normal behavior of the server and cannot be considered an error.

Reply:
I did this (and added the error type) because #77 requested it:

> If the method times out, it should return a TimeOut Exception.

If the team if good with returning None instead, I’ll do that.

Reply:
You are right. What do you think is best @curquiza?

Reply:
This is what I have so far for the native. It is passing the tests, but this is also my first time using the Futures crate so I have a few questions.

```rust
use std::thread::sleep;
use std::time::Duration;
use futures::channel::oneshot::{channel, Sender};
...

pub async fn wait_for_pending_update(
        &self,
        interval_ms: Option<Duration>,
        timeout_ms: Option<Duration>,
    ) -> Result<UpdateStatus, Error> {
    ...
                UpdateStatus::Enqueued { .. } => {
                    elapsed_time += interval;
                    let (sender, receiver) = channel::<()>();
                    let _thread = std::thread::spawn(move || async_sleep(sender, interval));
                    receiver.await.unwrap();
                },
    ...
}

#[cfg(not(target_arch = "wasm32"))]
pub(crate) fn async_sleep(sender: Sender<()>, interval: Duration) {
    sleep(interval);
    sender.send(()).unwrap();
}
```

1. Does this look correct, or would you expect something different?
2. Is it enough to send the sleep into the channel, or do I need to send the whole test?
2. Do you know of a way I can confirm that this isn't blocking?

Reply:
If it's more rust idiomatic, let's do the `None` :)

Reply:
(sorry for the delay)

Reply:
OK, what i ended up doing is wrapping the whole result in an Option. My thinking was you would want to be able to differentiate between the wait taking longer than specified and an actual server error.

If you all disagree with that and would rather always return None let me know and I'll change it.